webapp/src/utils/jira_issue_metadata.tsx

```diff
@@ -190,6 +190,47 @@ function isValidFieldForFilter(field: JiraField): boolean {
     (type === 'array' && allowedArrayTypes.includes(items));
 }
 
+export function getStatusField(metadata: IssueMetadata | null, issueTypes: string[]): FilterField | null {
+    // Filtering out the statuses on the basis of selected issue types
+    const issueTypesWithStatuses = metadata && metadata.issue_types_with_statuses;
+    const keys = new Set<string>();
+    const statuses: Status[] = [];
+    if (issueTypesWithStatuses) {
+        for (const issueType of issueTypesWithStatuses) {
+            if (issueTypes.includes(issueType.id) || !issueTypes.length) {
+                for (const status of issueType.statuses) {
+                    if (!keys.has(status.id)) {
+                        keys.add(status.id);
+                        statuses.push(status);
```

Comment:
Will `issueTypes` ever have length 0? (looking at `!issueTypes.length`)

Also it's a bit hard to read will all of the variations of:
- issueTypes
- issueTypesWithStatuses
- issueType
- issueType.statuses

Not sure there is a better way though. I was going to suggest renaming the function parameter `issueTypes` to `issueTypesWithoutStatuses`, but that sounds like "these issue types have no statuses associated with them". Current code seems fine, just a lot of redundancy causing a bit of confusion for me.

Reply:
> Will issueTypes ever have length 0? (looking at !issueTypes.length)
@mickmister Yes, when we have not selected any value in the modal.

Reply:
Maybe we can change the name to `selectedIssueTypes` to convey they are user set values

Reply:
> I will rename issueTypes to selectedIssueTypes for more clarity.
@mickmister Already changed